src/public_key.rs

```diff
@@ -240,6 +240,53 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn duplicate_key() {
+        let docker = clients::Cli::default();
+        let image = images::postgres::Postgres::default().with_version(9);
+        let container = docker.run(image);
+        let pool = setup_postgres(&container).await;
+        let docker = clients::Cli::default();
+        let image = images::postgres::Postgres::default().with_version(9);
+        let container = docker.run(image);
+        let pool = setup_postgres(&container).await;
+        let public_key_service = PublicKeyGrpc { db_pool: Arc::new(pool) };
+        let request1 = PublicKeyRequest {
+            public_key: Some(PublicKey {
+                key: Some(Key::Secp256k1(vec![1u8, 2u8, 3u8])),
+            }),
+            url: "http://test.com".to_owned(),
+            metadata: None,
+        };
+        let request2 = PublicKeyRequest {
+            public_key: Some(PublicKey {
+                key: Some(Key::Secp256k1(vec![1u8, 2u8, 3u8])),
+            }),
+            url: "http://test2.com".to_owned(),
+            metadata: None,
+        };
+
+        let (uuid, url) = match public_key_service.add(Request::new(request1)).await {
+            Ok(result) => {
+                let result = result.into_inner();
+                assert!(!result.uuid.is_none());
+                (result.uuid, result.url)
+            },
+            _ => unreachable!(),
+        };
+
+        match public_key_service.add(Request::new(request2)).await {
+            Ok(result) => {
+                let result = result.into_inner();
+                assert!(!result.uuid.is_none());
+                assert!(result.uuid == uuid);
```

Comment:
for these three asserts you can use `assert_eq!` and `assert_ne!`. It will help when they fail because it will show the to values instead of `expected true, actual false`.

Reply:
oh nice, yeah I'll do that... I need to get my VSCode Intellisense working better so I can see what options are available 😆